src/tunnel.c

```diff
@@ -95,7 +95,8 @@ static int pppd_run(struct tunnel *tunnel)
 			"/usr/sbin/pppd", "38400", "noipdefault", "noaccomp",
 			"noauth", "default-asyncmap", "nopcomp", "receive-all",
 			"nodefaultroute", ":1.1.1.1", "nodetach",
-			"lcp-max-configure", "40", "usepeerdns", "mru", "1354",
+			"lcp-max-configure", "40", "mru", "1354",
+			tunnel->config->pppd_use_peerdns ?  "usepeerdns" : NULL,
```

Comment:
On Fri, Aug 12, 2016 at 12:33:33AM -0700, Adrien Verg� wrote:
> > @@ -95,7 +95,8 @@ static int pppd_run(struct tunnel *tunnel)
> >  			"/usr/sbin/pppd", "38400", "noipdefault", "noaccomp",
> >  			"noauth", "default-asyncmap", "nopcomp", "receive-all",
> >  			"nodefaultroute", ":1.1.1.1", "nodetach",
> > -			"lcp-max-configure", "40", "usepeerdns", "mru", "1354",
> > +			"lcp-max-configure", "40", "mru", "1354",
> > +			tunnel->config->pppd_use_peerdns ?  "usepeerdns" : NULL,
> 
> Furthermore, the current implementation would break the `--pppd-log` and `--pppd-plugin` options (see the `i` variable).

OK, I'll look into this later today.

I also want to add syslog support. Do you want me to make this a
compile-time option (with runtime config option for turning it on) or
just runtime?

Ralf
-- 
Dr. Ralf Schlatterbeck                  Tel:   +43/2243/26465-16
Open Source Consulting                  www:   http://www.runtux.com
Reichergasse 131, A-3411 Weidling       email: office@runtux.com